source/server-selection/server-selection.rst

```diff
@@ -730,33 +730,35 @@ the command and how it is invoked:
 
       The current list of "may-use-secondary" commands includes:
 
-        - aggregate without a write stage (e.g. ``$out``, ``$merge``)
-        - collStats
-        - count
-        - dbStats
-        - distinct
-        - find
-        - geoNear
-        - geoSearch
-        - group
-        - mapReduce where the ``out`` option is ``{ inline: 1 }``
-        - parallelCollectionScan
+      - aggregate without a write stage (e.g. ``$out``, ``$merge``)
+      - collStats
+      - count
+      - dbStats
+      - distinct
+      - find
+      - geoNear
+      - geoSearch
+      - group
+      - mapReduce where the ``out`` option is ``{ inline: 1 }``
+      - parallelCollectionScan
 
       Associated command-specific helpers SHOULD take a read preference
       argument and otherwise MUST use the default read preference from client,
-      database or collection configuration.
-
-      The aggregate command succeeds on a secondary unless a write stage (e.g.
-      ``$out``, ``$merge``) is specified. When a write stage is specified, the
-      command must be treated as a write command. If the read preference is not
-      'primary', the driver SHOULD warn if a write stage is included in the
-      pipeline.
-
-      If a client provides a specific helper for inline mapreduce, then it is
-      "may-use-secondary" and the *regular* mapreduce helper is "must use
-      primary". Otherwise mapreduce behaves like the aggregate helper: it is the
-      user's responsibility to specify {inline: 1} when running mapreduce on a
-      secondary.
+      database, or collection configuration.
+
+      For pre-5.0 servers, an aggregate command is "must-use-primary" if its
+      pipeline contains a write stage (e.g. ``$out``, ``$merge``); otherwise, it
+      is "may-use-secondary". For 5.0+ servers, secondaries can execute an
+      aggregate command with a write stage and all aggregate commands are
+      "may-use-secondary". This is discussed in more detail in
+      `Read preferences and server selection <../crud/crud.rst#read-preferences-and-server-selection>`_
+      in the CRUD spec.
```

Comment:
For the record, I haven't found these terms (e.g. "must-use-primary") used in other specs, despite the text below:

> Command helper specifications SHOULD use those terms for clarity.

Because this feature may require some deviation to the normal rules for server selection, I also opted not to introduce these terms in my CRUD spec additions.

I'm not even sure this paragraph makes sense in the scope of the server selection spec, because it's referring to server versions as an influencing factor of the command helper's type (e.g. "must-use-primary", "may-use-secondary"). Rather than try to explain that here, I thought it was easier to consolidate everything in the CRUD spec.

If this issue comes up down the line with _another_ helper, I suppose we can revisit and restructure the server selection spec. For now, I think we can avoid doing so in the spirit of "defy augury".